libs/dyn/dynvar/ref.go

```diff
@@ -30,23 +33,32 @@ type ref struct {
 //   - "${a.b}"
 //   - "${a.b.c}"
 //   - "${a} ${b} ${c}"
-func newRef(v dyn.Value) (ref, bool) {
+func newRef(v dyn.Value) (ref, ref, bool) {
```

Comment:
Should we have a separate function that matches `v` against a list of known failure modes? Something like:
```
func isMalformedRef(v dyn.Value} {
  if not matches "potentialVarRef" then return early.

  switch {
  case strings.Contains(v, "__"): return some error
  case strings.Contains(v, ".."): return some error
  default:
     return generic error that the ref is malformed
}
```

The signature of `newRef` gets a bit weird if we start returning the potential refs as well.